addons/mail/static/src/core/common/link_preview_confirm_delete.js

```diff
@@ -28,7 +28,10 @@ export class LinkPreviewConfirmDelete extends Component {
     onClickOk() {
         rpc(
             "/mail/link_preview/hide",
-            { link_preview_ids: [this.props.linkPreview.id] },
+            {
+                message_id: this.props.message.id,
+                link_preview_ids: [this.props.linkPreview.id]
```

Comment:
We can't get rid of the message in this component since we need it to delete all the link previews at once (delete all button).

But the link preview component doesn't depend of message anymore.

Reply:
Provide a `deleteAll` props, same idea as `delete`?

Reply:
I thought of that, but it would just be moving the code somewhere else, you end up with a weird props `deleteAll` props instead of `message`.
Since the message is optional, it doesn't change much, unless we think there is a use case outside of message that needs the "delete all" button?

Reply:
The goal is just to make the component as generic enough, as the model is fully split from message now.

If there is a message props, even optional, it's not split from message, and message specific logic is still present in that component code (in particular the component is responsible to determine how a link preview should be removed from a message). This code is best left on message I think.